common/connectors-config/src/main/java/com/vmware/connectors/common/utils/CommonUtils.java

```diff
@@ -0,0 +1,32 @@
+package com.vmware.connectors.common.utils;
+
+import com.vmware.connectors.common.payloads.response.Card;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpHeaders;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Common utility functions to be used across connectors parent.
+ */
+public final class CommonUtils {
+
+    public static final String IMAGE_PATH = "/images/connector.png";
+
+    private CommonUtils() {
+        // Singleton class.
+    }
+
+    public static void buildImageUrl(final Card.Builder card, final HttpServletRequest request) {
```

Comment:
Please rework this to reduce duplication and allow a better contract (you can keep the Card.Builder one as an overload if you want):
* https://github.com/vmware/connectors-workspace-one/blob/cc60cf58be7828d8ab24c8a9c52bff0620ef0103/common/connectors-config/src/main/java/com/vmware/connectors/common/web/ConnectorRootController.java#L74

A better approach is:

```java
public static String buildImageUrl(HttpServletRequest request) {
    return buildImageUrl(request, DEFAULT_IMAGE_PATH);
}

public static String buildImageUrl(HttpServletRequest request, String imagePath) {
    return UriComponentBuilder.fromHttpRequest(request).path(imagePath).build().toUriString();
}

// if you want to keep the card builder overload
public static void buildImageUrl(Card.Builder card, HttpServletRequest request) {
    card.setImageUrl(buildImageUrl(request));
}


// and in the Discovery code:
if (imageResource.exists()) {
    String image = CommonUtils.buildImageUrl(request);
    resource.add(new Link(image, "image"));
}
```

That is more flexible (what if someone wants an image url for something that doesn't involve building a card -- like the discovery code) and doesn't force the caller to have to break up their builder method chains if they don't want to.

Reply:
I have added overloaded methods for ```buildConnectorImageUrl```. We can't use ```UriComponentBuilder.fromHttpRequest``` here as the original request was ```/card/requests```. Image url will be ```http://bitbucket-connector/card/requests/images/connector.png```. I retained the logic to build the image url.

Reply:
You can use fromHttpRequest and replacePath.  There is an example in the mock-connectors repo I believe:

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/util/UriComponentsBuilder.html#replacePath-java.lang.String-

Reply:
Replaced the logic using ```UriComponentsBuilder.fromHttpRequest(request).replacePath(path).build().toString()```.